internal/ldap/dn.go

```diff
@@ -181,6 +229,29 @@ func (d *DN) Equal(other *DN) bool {
 	return true
 }
 
+// RDNsMatch returns true if the individual RDNs of the DNs
+// are the same regardless of ordering.
+func (d *DN) RDNsMatch(other *DN) bool {
+	if len(d.RDNs) != len(other.RDNs) {
```

Comment:
This one is to check that they should have the same number of elements, if they don't then they are considered different already.  Otherwise continue to check that the they both have the same RDNs with the same attributes.

Reply:
Cool, every entry has to match then